src/interface/misc.rs

```diff
@@ -215,7 +215,7 @@ pub fn new_hashmap<K: std::cmp::Eq + std::hash::Hash, V>() -> RustHashMap<K, V>
 }
 
 pub unsafe fn charstar_to_ruststr<'a>(cstr: *const i8) -> Result<&'a str, Utf8Error> {
-    return std::ffi::CStr::from_ptr(cstr).to_str();         //returns a result to be unwrapped later
+    return std::ffi::CStr::from_ptr(cstr as *const u8).to_str();         //returns a result to be unwrapped later
```

Comment:
We're going to start using using [conditional compilation](https://doc.rust-lang.org/reference/conditional-compilation.html) which is a new venture for us. Apparently this needs to be u8 for mac but i8 for linux? It might be worth factoring out this function and doing the condtional outside of here?